mlflow/pyfunc/__init__.py

```diff
@@ -266,7 +270,139 @@ def _load_model_env(path):
     return _get_flavor_configuration(model_path=path, flavor_name=FLAVOR_NAME).get(ENV, None)
 
 
-def load_model(model_uri, suppress_warnings=True):
+def _enforce_type(name, values: pandas.Series, t: DataType):
+    """
+    Enforce the input column type matches the declared in model input schema.
+
+    The following type conversions are allowed:
+
+    1. np.object -> string
+    2. int -> long
+    3. float -> double.
+
+    Any other type mismatch will raise error.
+    """
+    if values.dtype == np.object and t not in (DataType.binary, DataType.string):
+        values = values.infer_objects()
+
+    if values.dtype in (t.to_pandas(), t.to_numpy()):
+        # The types are already compatible => conversion is not necessary.
+        return values
+
+    if t == DataType.binary and values.dtype.kind == t.binary.to_numpy().kind:
+        #  NB: bytes in numpy have variable itemsize and would fail the upcast check.
+        return values
+
+    if t == DataType.string and values.dtype == np.object:
+        #  NB: strings are by default parsed and inferred as objects, but it is
+        # recommended to use StringDtype extension type if available. See
+        #
+        # `https://pandas.pydata.org/pandas-docs/stable/user_guide/text.html`
+        #
+        # for more detail.
+        try:
+            return values.astype(t.to_pandas(), errors="raise")
+        except ValueError:
+            raise MlflowException(
+                "Failed to convert column {0} from type {1} to {2}.".format(
+                  name, values.dtype, t)
+            )
+
+    numpy_type = t.to_numpy()
+    is_compatible_type = values.dtype.kind == numpy_type.kind
+    is_upcast = values.dtype.itemsize <= numpy_type.itemsize
+    if is_compatible_type and is_upcast:
+        # upcasting compatible types is ok
```

Comment:
the above if statement has great variable names, so no need for this comment :)

Reply:
:)